docsite/source/predicates.html.md

```diff
@@ -4,104 +4,676 @@ layout: gem-single
-
-``` ruby
-require 'dry/logic'
-require 'dry/logic/predicates'
-
-include Dry::Logic
-
-is_hash = Rule::Predicate.new(Predicates[:type?]).curry(Hash)
-# => #<Dry::Logic::Rule::Predicate predicate=#<Method: Module(Dry::Logic::Predicates::Methods)#type?> options={:args=>[:hash]}>
-name_key = Rule::Predicate.new(Predicates[:key?]).curry(:name)
-# => #<Dry::Logic::Rule::Predicate predicate=#<Method: Module(Dry::Logic::Predicates::Methods)#key?> options={:args=>[:name]}>
-
-hash_with_key = is_hash & name_key
-# => #<Dry::Logic::Operations::And rules=[#<Dry::Logic::Rule::Predicate predicate=#<Method: Module(Dry::Logic::Predicates::Methods)#type?> options={:args=>[:hash]}>, #<Dry::Logic::Rule::Predicate predicate=#<Method: Module(Dry::Logic::Predicates::Methods)#key?> options={:args=>[:name]}>] options={}>
-
-hash_with_key.(name: 'John').success?
-# => true
-
-hash_with_key.(not_valid: 'John').success?
-# => false
-
-hash_with_key.([1,2]).success?
-# => false
+## Respond to
+
+> See `attr?`.
+
+``` ruby
+respond_to?(:name)
+```
+
+``` ruby
+# Pass
+Struct.new(:name).new("John")
+
+# Fail
+Struct.new(:age).new(100)
+```
+
+## URI
+
+> Verify user input is a URL.
+
+``` ruby
+uri?(:https)
+uri?(:http)
+uri?(/https?/)
+uri?([:http, :https])
+```
+
+``` ruby
+# Pass
+"https://google.com"
+
+# Fail
+"localhost"
+```
+
+## UUID (1-5)
+
+> Verify user input against UUID 1-5
+
+``` ruby
+uuid_v1?
+uuid_v2?
+uuid_v3?
+uuid_v4?
+uuid_v5?
+```
+
+``` ruby
+# Pass UUID-1
+"554ef240-5433-11eb-ae93-0242ac130002"
+
+# Fail
+"localhost"
+```
+
+## Case expression
+
+> Implements the `===` operator
+
+``` ruby
+case?(5..10)
+case?(Integer)
+```
+
+``` ruby
+# Pass
+10
+1
+
+# Fail
+100
+"string"
+```
+
+## True / False
+
+> Check for a boolean value
+
+``` ruby
+true?
+false?
+```
+
+``` ruby
+# Pass
+true
+false
+
+# Fail
+false
+true
+```
+
+## Identity equality
+
+> Compare two values using `object_id`
+
+``` ruby
+is?(nil)
+is?(Class.new)
+```
+
+``` ruby
+# Pass
+nil
+
+# Fail
+Class.new
+```
+
+## Inequality
+
+``` ruby
+not_eql?(10)
+```
+
+``` ruby
+# Pass
+20
+
+# Fail
+10
+```
+
+## Equality
+
+> Implements Rubys compare operator `==` or `eql?`
+
+``` ruby
+eql?(10)
+```
+
+``` ruby
+# Pass
+10
+
+# Fail
+20
+```
+
+## Excluded values
+
+> Check for exclusion. Can be used on all values responding to `include?`
+
+``` ruby
+excludes?("A")
+excludes?(5)
+```
+
+``` ruby
+# Pass
+"BBB"
+[1, 2, 3]
+
+# Fail
+"AAA"
+[3, 4, 5]
+```
+
+## Included values
+
+> Check for inclusion. Can be used on all values responding to `include?`
+
+``` ruby
+includes?("A")
+includes?(5)
+```
+
+``` ruby
+# Pass
+"AAA"
+[3, 4, 5]
+
+# Fail
+"BBB"
+[1, 2, 3]
+```
+
+## Excluded from
+
+``` ruby
+excluded_from?([1, 2, 3])
+```
+
+``` ruby
+# Pass
+4
+
+# Fail
+1
+```
+
+## Included in
+
+``` ruby
+included_in?([1, 2, 3])
+```
+
+``` ruby
+# Pass
+1
+
+# Fail
+4
+```
+
+## Max byte size
+
+``` ruby
+max_bytesize?(1)
```

Comment:
Should we group similar methods under the same header, i.e `bytesize?`, `max_bytesize?` and `min_bytesize?`. That would make it easier to specify input types and method a description.

Reply:
The order seems to be more important for readability. So in this case `bytesize?`, `min_bytesize?` and `max_bytesize?`. Started moving them around in 49858b0.